### ISSUE TITLE
It would be nice to be able to calibrate the power management

### ISSUE THREAD

**Issue by naelolaiz** [FastLED/FastLED#511]
I'm testing the library and so far it works great, I love it. Thank you!
But reaching the power management functions I measured that the real current does not correspond to what I asked for (the diff was of 100mA). Would it be possible to add some way to enter empiric measures of the used device in order to calibrate the power management?

**Comment by marmilicious** [comment]
Could you use a copy of
https://github.com/FastLED/FastLED/blob/master/power_mgt.cpp
for your project and edit lines 25-28 to values closer to what you're reading?

**Comment by marmilicious** [comment]
You're right though, it would be nice to be able to define something at the top of your code for this.

**Comment by zackees** [comment]
Closing as **completed** — this shipped in 3.10.4.

You asked for "some way to enter empiric measures of the used device in order to calibrate the power management." That is now `FastLED.setPowerModel()`:

```cpp
// Measured on your own hardware: red/green/blue mW at full brightness,
// plus the per-LED idle draw.
FastLED.setPowerModel(PowerModelRGB(100, 48, 100, 12));
```

`PowerModelRGB` (`src/power_mgt.h:27`) carries exactly the four numbers the old hardcoded constants held, so measuring your strip and typing the results in is now a one-liner instead of @marmilicious's workaround of vendoring `power_mgt.cpp` and editing lines 25-28.

There is also a fifth parameter that didn't exist in any form back then:

```cpp
FastLED.setPowerModel(PowerModelRGB(40, 40, 40, 2, 0.87f));
```

`exponent` models a non-linear brightness-to-power response — 1.0 is linear (the default and the old behavior), below 1.0 means higher-than-linear draw at mid brightness. Real strips are usually not linear, so if your measurements disagree with FastLED at *some* brightnesses but not others, that knob is likely what you want rather than the channel weights.

Relevant to your specific 100 mA discrepancy: the source has long carried an alternate calibration from RAtkins measured on the **PSU input** side rather than the LED side —

```
red 100, green 48, blue 100, dark 12
```

— noted as "probably about 20%-25% too high due to PSU heat losses, but if you're measuring wattage on the PSU input side, this may be a better set of calibrations." If that's where your meter was, that set is now something you can just pass to `setPowerModel()`.

Landed in `3785a40f` (2026-01-02) via #2152, with tests in `tests/power_mgt.cpp`.

**One caveat, so nobody is surprised:** `PowerModelRGBW` and `PowerModelRGBWW` exist as types but currently forward to `PowerModelRGB` and ignore the white channels — see the `@note` at `src/power_mgt.h:57`. So calibration is complete for 3-channel strips and incomplete for RGBW. If you need RGBW power calibration, that's worth a fresh issue.

Thanks @marmilicious for the interim workaround that carried people for eight years.
